mad-dog/src/tests/test_userReplicaSetManager.js

```diff
@@ -0,0 +1,278 @@
+    let primary = usrReplicaInfoFromContract.primaryId
+    let secondaries = usrReplicaInfoFromContract.secondaryIds
+
+    let newPrimary = parseInt(secondaries[0])
+    let newSecondaries = [primary, secondaries[1]].map(x=>parseInt(x))
+    logger.info(`userId: ${userId} | P: ${primary}->${newPrimary}`)
+    logger.info(`userId: ${userId} | S1: ${secondaries[0]}->${newSecondaries[0]}`)
+    logger.info(`userId: ${userId} | S2: ${secondaries[1]}->${newSecondaries[1]}`)
+    let tx = await libs.updateReplicaSet(userId, newPrimary, newSecondaries)
+    await waitForBlock(libs, tx.blockNumber)
+  })
+}
+
+// Promote each user's secondary2 to primary
+// Replica set transitions: (P=Primary, S1=Secondary1, S2 = Secondary2)
+// P->S2, S1->S1, S2->P
+const promoteSecondary2ToPrimary = async(executeAll) => {
+  await executeAll(async (libs, i) => {
+    // Retrieve user id if known from walletIndexToUserIdMap
+    // NOTE - It might be easier to just create a map of wallets instead of using 'index'
+    const userId = walletIndexToUserIdMap[i]
+    let usrReplicaInfoFromContract = await libs.getUserReplicaSet(userId)
+    logger.info(`userId: ${userId}: promoteSecondary2ToPrimary`)
+    let primary = usrReplicaInfoFromContract.primaryId
+    let secondaries = usrReplicaInfoFromContract.secondaryIds
+
+    let newPrimary = parseInt(secondaries[1])
+    let newSecondaries = [secondaries[0], primary].map(x=>parseInt(x))
+    logger.info(`userId: ${userId} | P: ${primary}->${newPrimary}`)
+    logger.info(`userId: ${userId} | S1: ${secondaries[0]}->${newSecondaries[0]}`)
+    logger.info(`userId: ${userId} | S2: ${secondaries[1]}->${newSecondaries[1]}`)
+    let tx = await libs.updateReplicaSet(userId, newPrimary, newSecondaries)
+    await waitForBlock(libs, tx.blockNumber)
+  })
+}
+
+// Promote each user's secondary2 to primary
+// Replica set transitions: (P=Primary, S1=Secondary1, S2 = Secondary2)
+// P->P, S1->S2, S2->S1
+const swapSecondaries = async(executeAll) => {
+  await executeAll(async (libs, i) => {
+    // Retrieve user id if known from walletIndexToUserIdMap
+    // NOTE - It might be easier to just create a map of wallets instead of using 'index'
+    const userId = walletIndexToUserIdMap[i]
+    let usrReplicaInfoFromContract = await libs.getUserReplicaSet(userId)
+    logger.info(`userId: ${userId}: swapSecondaries`)
+    let primary = usrReplicaInfoFromContract.primaryId
+    let secondaries = usrReplicaInfoFromContract.secondaryIds
+    let newPrimary = primary
+    let newSecondaries = [secondaries[1], secondaries[0]].map(x=>parseInt(x))
+    logger.info(`userId: ${userId} | P: ${primary}->${newPrimary}`)
+    logger.info(`userId: ${userId} | S1: ${secondaries[0]}->${newSecondaries[0]}`)
+    logger.info(`userId: ${userId} | S2: ${secondaries[1]}->${newSecondaries[1]}`)
+    let tx = await libs.updateReplicaSet(userId, newPrimary, newSecondaries)
+    await waitForBlock(libs, tx.blockNumber)
```

Comment:
verify the signer here

Reply:
nvm, this is already done up here https://github.com/AudiusProject/audius-protocol/pull/1141/files#diff-3d579dcaaec0479950c9d8be459360c63da1f094dd50c24aab97ef0c3a470d86R46